15.md

```diff
@@ -121,21 +120,26 @@ Fields that are not self-explanatory:
    - to calculate the total cost of shipping for an order, the user will choose a shipping option during checkout, and then the client must consider this costs:
      - the `base cost from the stall` for the chosen shipping option
      - the result of multiplying the product units by the `shipping costs specified in the product`, if any.
+- `type`: It determines whether a product is a simple product, a variable product, or a variation of a variable product. (i.e.: a shirt sold in a single colour is a simple product, a shirt sold in more than one colour is a variable product, and has variations). To link the variable product within its variations, the variation product event should contain an `a` tag pointing to the event coordinate of the variable product. 
```

Comment:
1. how is the merchant keeping the inventory. If he has 3 red shirts and 2 blue ones then the client will see one product with 5 or two different products
1. how does the merchant (or anyone) decide that the blue shirt is a variation of the red one or vice-versa? What does it happen if there are no more red shirts (the "simple" product)?

Reply:
Isn't this request just a "group by" functionality? 
So web clients can display:
 - in the `product list` multiple products as one
 - in the `product details` the different grouped products that the user can select from

Reply:
Products that are variations of a variable product have a tag:`a` in their tags pointing to the product of which they are a variation. So if a person sees the variable product in a stall, this being a variable product means that it has some variations so the client should query for events that point to the variable product, using a filter. Ideally once the user selects the product to go browse its product page, although this is up to the developers of each client and how they want to treat variable products. 

The quantity of each variation, as well as the price is specified in the event of each variation, so you can have 5 red shirts and 2 purple shirts and the purple one is a bit more expensive. 

You can also use the `"specs"` field in the variable product to give a preview of the different variations that can be found.

Reply:
- how is this different than a `group by`?
- the "variations" approach is much more tangled

Reply:
How can you implement 'group by' here?
I think the design is extensible, flexible and simple enough to bring in the feature of variable products and their variations.